### PR TITLE
Support change node-configs via k3s-upgrade container

### PR DESCRIPTION
**Pull request by MonzElmasry**
This pr adds support for the upgrade container to change the config file placed on the host on k3s nodes, then restarts k3s process

issue : https://github.com/rancher/rke2/issues/43

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -21,12 +21,20 @@ RUN if [ "${ARCH}" == "amd64" ]; then \
  && chmod +x /opt/k3s \
  && file /opt/k3s
 
+RUN set -x \
+ && apk --no-cache add curl \
```
Comment (briandowns) [inline Dockerfile:25]: Can you use the spacing used in the rest of the file?

File: Dockerfile [hunk 2]
```diff
@@ -21,12 +21,20 @@ RUN if [ "${ARCH}" == "amd64" ]; then \
...
+ && curl -fsSLO https://github.com/rancher/config-modifier/releases/download/v0.0.1/config-modifier-${ARCH} \
```
Comment (galal-hussein) [inline Dockerfile:26]: can we make the version an ARG as well

File: scripts/upgrade.sh [hunk 3]
```diff
@@ -102,6 +102,22 @@ verify_masters_versions() {
   done
 }
 
+replace_config_file(){
+KUBECTL_BIN="/opt/k3s kubectl"
+CONFIG_FILE="/host/etc/rancher/k3s/config.yaml"
+LABELS=$(${KUBECTL_BIN} get nodes $HOSTNAME  --show-labels | tail -n +2  | awk '{print $6}' )
+LIST=$(echo $LABELS | tr "," "\n" | tr "=" "-")
+echo $LIST
```
Comment (galal-hussein) [inline scripts/upgrade.sh:110]: is that necessary or thats for debugging purposes

File: scripts/upgrade.sh [hunk 4]
```diff
@@ -102,6 +102,22 @@ verify_masters_versions() {
...
+ls /host/opt/k3s
```
Comment (galal-hussein) [inline scripts/upgrade.sh:111]: same as above comment?

**Review by dweomer — APPROVED** [review]

File: scripts/upgrade.sh [hunk 5]
```diff
@@ -102,6 +102,22 @@ verify_masters_versions() {
   done
 }
 
+replace_config_file(){
+KUBECTL_BIN="/opt/k3s kubectl"
+CONFIG_FILE="/host/etc/rancher/k3s/config.yaml"
+LABELS=$(${KUBECTL_BIN} get nodes $HOSTNAME  --show-labels | tail -n +2  | awk '{print $6}' )
+LIST=$(echo $LABELS | tr "," "\n" | tr "=" "-")
```
Comment (brandond) [inline scripts/upgrade.sh:109]: I'm confused by the `tr "=" "-"` bit - won't this mangle the labels?

File: scripts/upgrade.sh [hunk 6]
```diff
@@ -102,6 +102,22 @@ verify_masters_versions() {
...
+echo $LIST
+ls /host/opt/k3s
+config-modifier --host-path $CONFIG_FILE --config-path /host/opt/k3s/ --preserved-entries "datastore-endpoint" --preserved-entries "token" --node-labels $(echo $LIST |  sed "s/ / --node-labels /g")
```
Comment (brandond) [inline scripts/upgrade.sh:112]: Won't this break if the label values have spaces in them?

**Review by brandond — CHANGES_REQUESTED** [review]

File: scripts/upgrade.sh (hunk 5 above)
Comment (MonzElmasry) [reply scripts/upgrade.sh]: i had this originally cause secrets are named with label names, and it cant include `=` in it

File: scripts/upgrade.sh [hunk 7]
```diff
@@ -102,6 +102,20 @@ verify_masters_versions() {
   done
 }
 
+replace_config_file(){
+  KUBECTL_BIN="/opt/k3s kubectl"
+  CONFIG_FILE="/host/etc/rancher/k3s/config.yaml"
+  LABELS=$(${KUBECTL_BIN} get nodes $HOSTNAME  --show-labels | tail -n +2  | awk '{print $6}' )
+  LIST=$(echo $LABELS | tr "," "\n" | tr "=" "-")
+  config-modifier --host-path $CONFIG_FILE --config-path /host/opt/k3s/ --preserved-entries "datastore-endpoint" --preserved-entries "token" --node-labels $(echo $LIST |  sed "s/ / --node-labels /g")
```
Comment (brandond) [inline scripts/upgrade.sh:110]:
```suggestion
  LABELS=$(${KUBECTL_BIN} get nodes $HOSTNAME -o 'template={{range $key, $value := .metadata.labels}}--node-labels "{{$key}}={{$value}}" {{end}}')
  config-modifier --host-path $CONFIG_FILE --config-path /host/opt/k3s/ --preserved-entries "datastore-endpoint" --preserved-entries "token" $LABELS
```

**Review by brandond — CHANGES_REQUESTED** [review]

**Pull request closed without merging**